### PR TITLE
Make message queue implementations pluggable

### PR DESCRIPTION
**Pull request by thll**
## Description

Creates distinct interfaces for the purpose of "reading from", "writing to" and "acknowledging message receipt on" a message queue.
Makes implementations pluggable by providing means to register additional implementations and activating them in the config file.

/jenkins-pr-deps Graylog2/graylog-plugin-enterprise-integrations#487

## Motivation and Context

Forward-port of https://github.com/Graylog2/graylog2-server/pull/10091

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by mpfz0r — APPROVED** [review]
Works and LGTM

File: graylog2-server/src/main/java/org/graylog2/plugin/journal/RawMessage.java [hunk 1]
```diff
@@ -260,8 +274,14 @@ public String toString() {
         return helper.toString();
     }
 
+    public Object getMessageQueueId() {
+        return messageQueueId;
+    }
     public long getJournalOffset() {
```
Comment (bernd) [inline graylog2-server/src/main/java/org/graylog2/plugin/journal/RawMessage.java:280]: Should we annotate this as `@Deprecated` as well?

**Review by bernd — CHANGES_REQUESTED** [review]
I tested this in my test environment without any issues. Great work @mpfz0r and @thll! Thank you!

Only one single question:

File: graylog2-server/src/main/java/org/graylog2/plugin/journal/RawMessage.java (hunk 1 above)
Comment (thll) [reply graylog2-server/src/main/java/org/graylog2/plugin/journal/RawMessage.java:285]: Yes, I agree.

**Review by bernd — APPROVED** [review]

**Pull request merged**